vault/token_store.go

```diff
@@ -1472,7 +1472,11 @@ func (ts *TokenStore) handleCreateCommon(
 			// this is a role, add default unless explicitly disabled.
 			if len(finalPolicies) == 0 {
 				finalPolicies = policyutil.SanitizePolicies(parent.Policies, localAddDefault)
-			} else {
+			} else if len(role.DisallowedPolicies) == 0 {
+				// We only do this if role.DisallowedPolicies is empty because
```

Comment:
That's a good point. I think this is part of the holdover from the major logic change in 0.6.4 that made having both allowed/disallowed not possible.